#166789955 Users should be able to view and edit their profiles

IsaiahRn opened the pull request:
#### What does this PR do?

- Users should be able to view and edit their profiles

#### Description of Task to be completed

- Add view and edit functionality

- Display input error validation

- Add tests

#### How should this be manually tested?

1. Run the following in your terminal
```
git clone https://github.com/andela/ah-92explorers-frontend.git
cd ah-92explorers-frontend
git checkout ch-setup-e2e-tests-166789946
npm install
npm run dev
```

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?
[#166789955](https://www.pivotaltracker.com/story/show/166789955)
#### Screenshots (if appropriate)

- View Updated User Profile
<img width="1174" alt="Screenshot 2019-08-25 23 35 35" src="https://user-images.githubusercontent.com/40357462/63656189-5ca4cb80-c791-11e9-92a8-cf42dd32616c.png">

- Edit User Profile
<img width="1421" alt="Screenshot 2019-08-22 22 12 26" src="https://user-images.githubusercontent.com/40357462/63546381-02560180-c52a-11e9-8190-6fe6d5023c60.png">

- Input validation alerts
<img width="1330" alt="Screenshot 2019-08-25 23 36 27" src="https://user-images.githubusercontent.com/40357462/63656250-c45b1680-c791-11e9-8c11-b9bcce9d1f22.png">

<img width="1324" alt="Screenshot 2019-08-25 23 36 57" src="https://user-images.githubusercontent.com/40357462/63656233-a7264800-c791-11e9-9946-6edf8fe21c2f.png">



#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

kodek-sleuth commented on src/redux/reducers/profile.js:
```diff
@@ -0,0 +1,39 @@
+  const { type, payload } = action;
+
+  switch (type) {
+    case GET_PROFILE:
```
By doing this Case statement without expecting to return something is not inadequate. Its like creating an if staement like this;
```
if (GET_PROFILE){

} 
Refactor this in a way that if you expect a case, it should return something like you did in the case block after 
```

IsaiahRn replied:
> By doing this Case statement without expecting to return something is not inadequate. Its like creating an if staement like this;
> 
> ```
> if (GET_PROFILE){
> 
> } 
> Refactor this in a way that if you expect a case, it should return something like you did in the case block after 
> ```

Thanks! I am going to work on that.

elemanhillary-zz reviewed (changes requested):
@IsaiahRn I have seen that you are doing a shallow test on the components, for example, me as a `TTL` i want to be able to see that you have simulated button clicks in your test for the component to prove that the `buttons` actually react to click `events` and one more thing test the whole component to test for some `HTML elements` existence comparing them with want you as a developer expects.

Hervera reviewed (changes requested):
You have too many commits

Hervera reviewed (changes requested):
I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

IsaiahRn commented:
> I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

@Hervera , I think that this is a problem related to your npm, you can try to look into that?

niyoceles reviewed (dismissed):
Good work

Hervera reviewed (changes requested):
Test your Navbar. It has `F` point on `coverage`

IsaiahRn committed:
feat(user-profile): view and edit user’s profile

- add view and edit profile functionality
- add input validation
- write some tests

[Finishes #166789955]

niyoceles commented:
> I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

@Hervera delete `node_modules` folder and `package-lock.json`, then run `npm install'

Hervera reviewed (dismissed):
Ready to be merged

niyoceles reviewed (dismissed):
Good job

IsaiahRn commented on src/components/Layout/navBar.js:
```diff
@@ -0,0 +1,32 @@
+          <div className="restOfNav">
+            <img src={search} alt="" className="navIcons navIconColor" />
+            <img src={notification} alt="" className="navIcons navIconColor" />
+            <img src={manIcon} alt="" className="navIcons manIcon" />
```
Use the actual profile picture

IsaiahRn commented on src/redux/actions/actionCreators/profile.js:
```diff
@@ -0,0 +1,74 @@
+  checkToken();
+  try {
+    const formData = new FormData();
+    formData.append('firstName', profileData.firstName);
```
Remove code repetition

elemanhillary-zz reviewed (approved):
ready for merge
